lib/engine/operator.rb

```diff
@@ -60,5 +61,37 @@ def unplaced_tokens
     def placed_tokens
       @tokens.select(&:city)
     end
+
+    def current_book_value
+      trains = @trains.sum(&:price)
+      tokens = if @token_book_value_override
+                 @tokens.size * @token_book_value_override
+               else
+                 @tokens.filter(&:used).sum(&:price)
```

Comment:
don't use filter, use reject/select

Reply:
As in the other comment, I think it's possible to do just one iteration here as well:

```suggestion
                 @tokens.sum { |t| t.used ? t.price : 0 }
```